src/deadline/cinema4d_submitter/cinema4d_render_submitter.py

```diff
@@ -239,6 +239,16 @@ def _get_job_template(
 
 def _show_submitter(parent=None, f=Qt.WindowFlags()):
 
+    if Scene.changed():
+        prompt = QtWidgets.QMessageBox.warning(
+            None,
+            "Deadline Cloud",
+            "Save scene changes before submission?",
+            QtWidgets.QMessageBox.Ok | QtWidgets.QMessageBox.Cancel
+        )
+        if prompt == QtWidgets.QMessageBox.Ok:
+            Scene.save()
```

Comment:
Should we `return` here if the user picks cancel?

Looking back at the source issue (#53), the same problem would occur:

>   I had never saved a scene so the job attachments failed to upload anything which resulted in a "scene does not exist" error.

Another possible outcome is that the job is submitted with the previously-saved copy of the scene while the scene has unsaved changes.

Reply:
Oh, thanks for surfacing that Maya behaves this way. Consistency aside, I'm trying to think of scenarios where that behavior makes sense.

What are your thoughts?

Reply:
Hi @jusiskin  

FYI - The Houdini submitter also follows this warning style UX flow. 

I think guiding the user to save with a warning is better than forcing the user to save. Sometimes saving a big scene can take a long time and the user knows the changes since last save will have no effect on the result so they want to continue on to submit without further friction. Another issue is GetChanged() can return false positives in some cases and the user wants to skip past the save. 

From what I can tell most other render farm products use this flow so it shouldn't surprise the typical DCC user.

It's ultimately a subjective user experience preference about what the product should value more: strictness or flexibility.

I'm happy to revise the code based on consensus of the maintainers. Thanks!

Reply:
@gp-dan that makes sense, thanks for the context. I also realize that this function only displays the submitter dialog and doesn't submit it. So the person using this can still cancel their submission by closing the submitter dialog.

On that basis, this behavior seems okay to me.